crates/polars-plan/src/plans/conversion/type_coercion/binary.rs

```diff
@@ -47,55 +47,6 @@ fn is_cat_str_binary(type_left: &DataType, type_right: &DataType) -> bool {
     }
 }
 
-fn process_list_arithmetic(
```

Comment:
remove logic from the `type_coercion` optimizer - this should already be handled by `get_arithmetic_field` during conversion